File: tests/testdata/codegen/aws-static-website-pp/dotnet/aws-static-website.cs
using System.Collections.Generic;
using System.Linq;
using Pulumi;
using AwsStaticWebsite = Pulumi.AwsStaticWebsite;

return await Deployment.RunAsync(() => 
{
    var websiteResource = new AwsStaticWebsite.Website("websiteResource", new()
    {
        SitePath = "string",
        IndexHTML = "string",
        CacheTTL = 0,
        CdnArgs = new AwsStaticWebsite.Inputs.CDNArgsArgs
        {
            CloudfrontFunctionAssociations = new[]
            {
                new Aws.Cloudfront.Inputs.DistributionOrderedCacheBehaviorFunctionAssociationArgs
                {
                    EventType = "string",
                    FunctionArn = "string",
                },
            },
            ForwardedValues = new Aws.Cloudfront.Inputs.DistributionDefaultCacheBehaviorForwardedValuesArgs
            {
                Cookies = new Aws.Cloudfront.Inputs.DistributionDefaultCacheBehaviorForwardedValuesCookiesArgs
                {
                    Forward = "string",
                    WhitelistedNames = new[]
                    {
                        "string",
                    },
                },
                QueryString = false,
                Headers = new[]
                {
                    "string",
                },
                QueryStringCacheKeys = new[]
                {
                    "string",
                },
            },
            LambdaFunctionAssociations = new[]
            {
                new Aws.Cloudfront.Inputs.DistributionOrderedCacheBehaviorLambdaFunctionAssociationArgs
                {
                    EventType = "string",
                    LambdaArn = "string",
                    IncludeBody = false,
                },
            },
        },
        CertificateARN = "string",
        Error404 = "string",
        AddWebsiteVersionHeader = false,
        PriceClass = "string",
        AtomicDeployments = false,
        Subdomain = "string",
        TargetDomain = "string",
        WithCDN = false,
        WithLogs = false,
    });

});

File: tests/testdata/codegen/aws-static-website-pp/go/aws-static-website.go
package main

import (
	awsstaticwebsite "github.com/pulumi/pulumi-aws-static-website/sdk/go/aws-static-website"
	"github.com/pulumi/pulumi-aws/sdk/v5/go/aws/cloudfront"
	"github.com/pulumi/pulumi/sdk/v3/go/pulumi"
)

func main() {
	pulumi.Run(func(ctx *pulumi.Context) error {
		_, err := awsstaticwebsite.NewWebsite(ctx, "websiteResource", &awsstaticwebsite.WebsiteArgs{
			SitePath:  pulumi.String("string"),
			IndexHTML: "string",
			CacheTTL:  0,
			CdnArgs: &*awsstaticwebsite.CDNArgsArgs{
				CloudfrontFunctionAssociations: []cloudfront.DistributionOrderedCacheBehaviorFunctionAssociationArgs{
					{
						EventType:   pulumi.String("string"),
						FunctionArn: pulumi.String("string"),
					},
				},
				ForwardedValues: &*cloudfront.DistributionDefaultCacheBehaviorForwardedValuesArgs{
					Cookies: &cloudfront.DistributionDefaultCacheBehaviorForwardedValuesCookiesArgs{
						Forward: pulumi.String("string"),
						WhitelistedNames: []pulumi.String{
							pulumi.String("string"),
						},
					},
					QueryString: pulumi.Bool(false),
					Headers: []pulumi.String{
						pulumi.String("string"),
					},
					QueryStringCacheKeys: []pulumi.String{
						pulumi.String("string"),
					},
				},
				LambdaFunctionAssociations: []cloudfront.DistributionOrderedCacheBehaviorLambdaFunctionAssociationArgs{
					{
						EventType:   pulumi.String("string"),
						LambdaArn:   pulumi.String("string"),
						IncludeBody: false,
					},
				},
			},
			CertificateARN:          "string",
			Error404:                "string",
			AddWebsiteVersionHeader: false,
			PriceClass:              "string",
			AtomicDeployments:       false,
			Subdomain:               "string",
			TargetDomain:            "string",
			WithCDN:                 false,
			WithLogs:                false,
		})
		if err != nil {
			return err
		}
		return nil
	})
}
